### PR TITLE
qt5* ports: additional fixes

### PR DESCRIPTION
**Pull request by chrstphrchvz**
###### Tested on

macOS 12.5
Xcode 14 beta 3

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Review by mascguy — APPROVED** [review]

File: aqua/qt55/Portfile
Comment (chrstphrchvz) [inline aqua/qt55/Portfile:145]: I have not ruled out whether anyone may have successfully built the bundled assimp, so incrementing the revision anyway.

**Review by mascguy — APPROVED** [review]
My only complaint, is that you haven't formally added yourself as a co-maintainer to the various qt5* (and qt6) ports yet!  ;-)

**Comment by mascguy** [comment]
Chris, in the interest of reducing the number of fix iterations, I'm willing to help pre-validate these changes. Particularly if there's only one macOS release where things are problematic - like 10.14 - where CI can't help.

**Comment by mascguy** [comment]
> Chris, in the interest of reducing the number of fix iterations, I'm willing to help pre-validate these changes. Particularly if there's only one macOS release where things are problematic - like 10.14 - where CI can't help.

That said, are you running on an Intel-based Mac? I ask, as Parallels provides great support for virtualizing older macOS releases. And I'd highly recommend it, if you're not running on ARM.

**Comment by chrstphrchvz** [comment]
I am not currently interested in adding myself as a maintainer of these ports. I just happened to have a moment to tinker with them. I likely need to first propose to put back `known_fail yes` before trying to fix any remaining failures.

I remain uninterested in having my own installs of older macOS, so I appreciate others’ interest in testing things for me.

**Comment by chrstphrchvz** [comment]
Inclined to revisit https://trac.macports.org/ticket/60975 and https://trac.macports.org/ticket/61850 . I think the original cause is that the build is using bison from PATH rather than /usr/bin/bison, which the patch and bison dependency works around. I suspect other ports besides qt5-qtwebkit and qt511-qtwebkit are affected.

**Comment by chrstphrchvz** [comment]
In f2f607688a48 `port:flex` was added as a build dependency qt5-qtwebkit. Presumably this should be consistently done for all qtwebkit ports, but I am not aware it makes any difference whether CLT flex or MacPorts flex is used.

**Comment by chrstphrchvz** [comment]
I believe only other thing I intended to add to this PR was more `known_fail yes` for qtwebengine ports on various macOS versions, but I haven’t sat down for long enough to make that happen. Otherwise this PR is at a stopping point and could be merged when the builders aren’t busy.

**Comment by catap** [comment]
@mascguy am I right that we must wait for 24 hours more before this one can be merge?

**Comment by mascguy** [comment]
> @mascguy am I right that we must wait for 24 hours more before this one can be merge?

Do you mean this PR for qt5* fixes?

**Comment by catap** [comment]
Yes, I do.

**Comment by mascguy** [comment]
> Yes, I do.

No, we can merge anytime, as it's been open since July 24th.

**Comment by catap** [comment]
When I'd like to ask you to do it :) Because I hope that ICU can be merged also, and this one is kind of a blocker for me.

Why? To not trigger rebuild of qt5x subports which son't be build with new bison which may leads to some tickets :)

**Review by mascguy — APPROVED** [review]
I only did a quick cursory review of the various patches, as there's a lot there.

But the portfile changes look fine. Merging.

**Comment by catap** [comment]
Thanks!

**Pull request merged**
